CSBDeep resources overview

frauzufall opened the pull request:
Hi @maweigert @uschmidt83 @fjug @tibuch @alex-krull,

I started a very rough, content-missing document to collect all CSBDeep resources, you should be able to preview it like this:

https://htmlpreview.github.io/?https://github.com/CSBDeep/CSBDeep_website/blob/howto/docs/howto/csbdeep.html

I would like to first collect all existing resources as links to this document, then probably split it up per tool etc. and in a next step it should be integrated into http://csbdeep.bioimagecomputing.com/. 

This means we should rethink the whole page. It needs to be simpler for people to understand how our tools are used officially.

If you agree, I would appreciate links to resources / missing topics / any other ideas.

maweigert commented:
Great idea, I'm all for it! Currently it's anyway a bit hard to navigate the different repos, so having a visually easy to navigate central landing page makes totally sense (e.g. doing so via the csbdeep github.io page?)

frauzufall commented:
I think https://csbdeep.bioimagecomputing.com/ is a good URL, we can split up the current content, have one landing page listing the resources and common questions like "how to get GPU support", and then subpages for the tools, and e.g. the current gallery would move to the CARE subpage.

frauzufall commented:
Ok, I went a bit ahead and made a new site based on the old design (just a test repo, will be copied here in the end):

https://frauzufall.github.io/csbdeep-testsite

I split the content into tools (CARE, N2V, StarDist) and scenarios. The idea is that a biologists do not need to first read and understand all about the different tools - they can directly follow a scenario matching or similar to their use case.  

There is quite some content missing:
- the tool HowTo's for [CARE](https://github.com/frauzufall/csbdeep-testsite/tree/master/_pages/tools/care) and [N2V](https://github.com/frauzufall/csbdeep-testsite/tree/master/_pages/tools/n2v)
- [scenario details](https://github.com/frauzufall/csbdeep-testsite/tree/master/_scenarios)

These files are markdown, so hopefully easy to edit.

For the tools, we can either add all the content or link to existing resources. E.g. N2V and CARE have subpages whereas for StarDist I just added one page now with the appropriate links. I was not sure if @maweigert and @uschmidt83 want to maintain this site or if they want to keep their documentation on their repos. You decide!

The current CARE scenario details are copied from the old gallery descriptions and don't make too much sense. 

The idea for the scenario description would be to write about data acquisition, data preparation, and roughly how the according tool was used. Ideally linking e.g. to a notebook / model in the model zoo / ...

If you add a scenario with `inputimage` and `outputimage` or `videos` (look at the existing scenarios like [Planaria denoising](https://github.com/frauzufall/csbdeep-testsite/blob/master/_scenarios/denoisingplanaria.md)), they will be automatically added to the gallery / to the videos of the tool the scenario is assigned to. I think that's quite cool, that means in the tool gallery you can read more about how each of the results was achieved by opening it's matching scenario.

Please have an opinion / let me know how you would like to contribute :)

frauzufall commented:
Notes from talking to @tibuch: 
- the landing page needs more content
- the tools can maybe be displayed in a flow chart (do you have image pairs? are they both noisy? ..)

uschmidt83 commented:
Hi, I think it’s a great idea to have a modern webpage that guides people to use appropriate methods based on their needs.

However, I’m not sure that using the name CSBDeep is such a good idea. I associate the python and Fiji package/plugin with it, not a more general umbrella term. Furthermore, Martin has moved to EPFL in Lausanne, Florian will move too in the foreseeable future. Also, it might be a good idea to include methods not developed by us in this flowchart.

frauzufall commented:
@uschmidt83 I agree. Much of the Java code will also migrate from CSBDeep/CSBDeep_fiji to imagej/imagej-modelzoo. But what would be the more general umbrella term?

frauzufall commented:
I am unsure about other methods there, because that will create more work to maintain the page and one has to decide what is added there and how and what isn't... That would be more like a DL in microscopy blog. I can't do all the necessary research to maintain something like this. I see it as a place to make sure the purpose and usability of the tools we develop and maintain is accessible. I can imagine keeping this up to date for the Jug Lab tools, having you @uschmidt83 and @maweigert and others from CSBD like @colemanbroad on board too would be great, since our work is interlinked anyways.
